multiboot2/src/builder/mod.rs

```diff
@@ -4,3 +4,41 @@ mod information;
 pub(self) mod traits;
 
 pub use information::Multiboot2InformationBuilder;
+
+use alloc::alloc::alloc;
+use alloc::boxed::Box;
+use core::alloc::Layout;
+use core::mem::size_of;
+
+use crate::{TagTrait, TagTypeId};
+
+/// Create a boxed tag with the given content.
+pub(super) fn boxed_dst_tag<T>(typ: TagTypeId, content: &[u8]) -> Box<T>
+where
+    T: TagTrait + ?Sized + ptr_meta::Pointee<Metadata = usize>,
```

Comment:
nit: I think you can also do sth like `TagTrait<Metadata = usize>` as `TagTrait` depends on `Pointee`.